### PR TITLE
docs: add documentation for configuring the Let's Encrypt cert

### LINKED ISSUE #124: Letsencrypt examples
Give examples of letsencrypt usage

### PR DESCRIPTION
**Pull request by jamaljsr**
Closes #124 

Added a `letsencrypt.md` doc which explains how to configure `litd` to run with a publicly verifiable TLS certificate provided by Let's Encrypt. It also provides examples for the `loop` and `frcli` CLI commands to communicate with the daemons.

This PR is based on the [grub-v2](https://github.com/lightninglabs/lightning-terminal/tree/grub-v2) branch (PR #116). I'll rebase after that PR is merged.

### CODE REVIEW

File: doc/letsencrypt.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# Terminal with a Let's Encrypt Certificate
+
+You can configure the HTTPS server to automatically install a free SSL certificate
+provided by [LetsEncrypt](https://letsencrypt.org/). This is recommended if you plan to
```
Comment (guggero) [inline doc/letsencrypt.md:4]: nit: should be "Let's Encrypt" here too.

File: doc/letsencrypt.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+access the website from a remote computer and do not want to deal with the browser warning
+you about the self-signed certificate. Another benefit is you can use a memorable URL to
```
Comment (guggero) [inline doc/letsencrypt.md:6]: nit: s/you about/you get about/

File: doc/letsencrypt.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+access the Terminal UI instead of an IP address.
+
+> Note: LiT only serves content over **HTTPS**. If you do not use `letsencrypt`, LiT will
+> use the self-signed certificate that is auto-generated by `lnd` to encrypt the
```
Comment (guggero) [inline doc/letsencrypt.md:10]: Worth adding "auto-generated by `lnd` (or LiT itself if `lnd` is running on a remote host)"?

File: doc/letsencrypt.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+- Purchase a domain name to use. It can be a root domain such as `mydomain.com` or a
+  sub-domain such as `terminal.mydomain.com`.
+- Update the domain's DNS to point the domain name to your node's public IP address, by
+  creating an `A` record. The specific steps to configure the `A` record varies depending
+  on the service you use to host your domain's DNS zone.
+
+## Firewall Configuration
+
+In order for LetsEncrypt to automatically install and renew your SSL certificate, it will
```
Comment (guggero) [inline doc/letsencrypt.md:24]: nit: Let's Encrypt

File: doc/letsencrypt.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+port `80` without needing to run the daemon as root.
+
+If you are able to use port-forwarding on your router/firewall, you can specify a
+different port (ex: `8080`) to listen for LetsEncrypt challenges using the
+`letsencryptlisten` flag.
+
+## LiT Configuration
+
+There are a few `litd` flags that you need to set to make use of LetsEncrypt certificates.
+These can be provided ont he command line or via the `lit.conf` file.
```
Comment (guggero) [inline doc/letsencrypt.md:44]: nit: s/ont he/on the/

File: doc/letsencrypt.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+| ------------------- | -------- | ----------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `letsencrypt`       | Yes      | `0`                     | Use Let's Encrypt to create a TLS certificate for the UI instead of using `lnd`'s TLS certificate.                                                                                                                                                                                                                                                                                                                                                 |
```
Comment (guggero) [inline doc/letsencrypt.md:48]: nit: should we use `false` instead of `0`?

File: doc/letsencrypt.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+**Examples:**
+
+Command Line:
+
+`$ litd --letsencrypt --letsencrypthost=terminal.mydomain.com`
+
+Configuration file (`litd.conf`):
+
+```
+--letsencrypt=1
```
Comment (guggero) [inline doc/letsencrypt.md:62]: nit: should use `true`

**Review by guggero — APPROVED** [review]
Very nice, thanks for the great addition to the doc!
Just a few nits, otherwise LGTM :1st_place_medal:

**Comment by jamaljsr** [comment]
@guggero thanks for the feedback and clarifications. I made the changes you suggested. 👍

**Review by justinpobrien — APPROVED** [review]
LGTM. Sorry for delay here I didn't realize I was requested as reviewer!

**Comment by guggero** [comment]
Can be rebased against master now that the dependent PR is in.

**Pull request merged**